### PR TITLE
switching regression and tool_script_runner

### PR DESCRIPTION
**Pull request by ue71603**
* tool_runner for regression switched.

### CODE REVIEW

**Comment by ue71603** [comment]
I moved the stuff. However, it seems either some things broke or new problems are uncovered.

As the script_runner now behaves as you want it, you can use it as well.

scripts/scripts_regression.txt contains the scripts use.

%%dir%% work directory for output
%%block%% the name of the script block (to generate subdirectories in the work directory)

With the new system I expect you to use it, so I will only go through those.

I will add all problems here. If we need to split them, we can do those.

###  python tool_script_runner.py ./scripts/scripts_regression.txt run.log nl1   
nothing is imported in netex_to_db.py, but I think this should not happen ;-)  I don't see what I have changed there that could have done that.  

### python tool_script_runner.py ./scripts/scripts_regression.txt run.log swissmikro

Somthing wents wrong in related_explorer.py
```
2024-12-11 23:55:51,716 - script_runner - INFO - test_runner: related_explorer.py with d:/aux_testing_processing/swissmikro/netex2-database.duckdb ServiceJourney random d:/aux_testing_processing/swissmikro/sj.xml False
2024-12-11 23:55:51,800 - script_runner - INFO - related_explorer: updating embedded and referencing tables
2024-12-11 23:55:51,800 - script_runner - INFO - related_explorer: updating embedded and referencing tables
2024-12-11 23:55:51,850 - script_runner - ERROR - Invalid Input Error: This function can not be called with an active transaction!, commit or abort the existing one first: Traceback (most recent call last):
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\tool_script_runner.py", line 194, in <module>
    main(args.script_file, args.log_file, args.log_level, args.blockname, args.begin_step)
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\tool_script_runner.py", line 154, in main
    result=load_and_run(script_name, script_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\tool_script_runner.py", line 20, in load_and_run
    result = mod(*args)
             ^^^^^^^^^^
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\related_explorer.py", line 184, in main
    embedding_update(db)
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\transformers\embedding.py", line 17, in embedding_update
    con.create_function('embedding', functools.partial(embedding_udf, db.serializer), return_type=list[str])
duckdb.duckdb.InvalidInputException: Invalid Input Error: This function can not be called with an active transaction!, commit or abort the existing one first

2024-12-11 23:55:51,850 - script_runner - ERROR - Invalid Input Error: This function can not be called with an active transaction!, commit or abort the existing one first: Traceback (most recent call last):
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\tool_script_runner.py", line 194, in <module>
    main(args.script_file, args.log_file, args.log_level, args.blockname, args.begin_step)
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\tool_script_runner.py", line 154, in main
    result=load_and_run(script_name, script_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\tool_script_runner.py", line 20, in load_and_run
    result = mod(*args)
             ^^^^^^^^^^
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\related_explorer.py", line 184, in main
    embedding_update(db)
  File "C:\Users\ue71603\MG_Daten\github\reference1\gtfs-netex-test\transformers\embedding.py", line 17, in embedding_update
    con.create_function('embedding', functools.partial(embedding_udf, db.serializer), return_type=list[str])
duckdb.duckdb.InvalidInputException: Invalid Input Error: This function can not be called with an active transaction!, commit or abort the existing one first


```

### python tool_script_runner.py ./scripts/scripts_regression.txt run.log blablacar
It seems to me that epip_db_to_xml.py is not terminating.
The previous script parts work fine.

**Comment by ue71603** [comment]
@skinkie  from the code I don't see, why this is not working

File: gtfs-netex-test/epip_db_to_xml.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def generator(self) -> Generator | None:
 
         return None
 
-def export_epip_network_offer(database_original, database_target, output_filename: str):
+def main(database_original, database_target, output_filename: str):
```
Comment (skinkie) [inline gtfs-netex-test/epip_db_to_xml.py:91]: Not what should be changed. Happy to make a main, but this is a generic function.

**Comment by skinkie** [comment]
Merged in master.

**Pull request closed without merging**